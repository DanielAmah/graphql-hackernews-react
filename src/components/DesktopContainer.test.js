/* eslint-disable */
import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import DesktopContainer from "./DesktopContainer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DesktopContainer />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("DesktopContainer component", () => {
  test("it matches the snapshot", () => {
    const component = create(<DesktopContainer />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("DesktopContainer hideFixedMenu method", () => {
  it("shows state change with hideFixedMenu function", () => {
    const component = create(<DesktopContainer />);
    const instance = component.getInstance();
    instance.hideFixedMenu();
    expect(instance.state.fixed).toBe(false);
  });
});

describe("DesktopContainer showFixedMenu method", () => {
  it("shows state change with showFixedMenu function", () => {
    const component = create(<DesktopContainer />);
    const instance = component.getInstance();
    instance.showFixedMenu();
    expect(instance.state.fixed).toBe(true);
  });
});
