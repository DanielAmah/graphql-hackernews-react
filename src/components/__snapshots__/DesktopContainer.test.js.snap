// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopContainer component it matches the snapshot 1`] = `
<div>
  <div>
    <div
      className="ui vertical center aligned segment"
      style={
        Object {
          "minHeight": 700,
          "padding": "1em 0em",
        }
      }
    >
      <div
        className="ui large pointing secondary menu"
      >
        <div
          className="ui container"
        >
          <a
            className="active item"
            onClick={[Function]}
          >
            Home
          </a>
          <div
            className="right item"
            onClick={[Function]}
          >
            <span>
              <a
                className="ui button"
                onClick={[Function]}
                role="button"
              >
                Log in
              </a>
              <a
                className="ui button"
                onClick={[Function]}
                role="button"
                style={
                  Object {
                    "marginLeft": "0.5em",
                  }
                }
              >
                Sign Up
              </a>
            </span>
          </div>
        </div>
      </div>
      <div>
        <div
          className="ui container"
        >
          <div
            className="ui clearing segment create-link-container"
          >
            <button
              className="ui button ui right floated teal button"
              id="handle-open"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              Add Link
            </button>
          </div>
          <div
            className="ui active centered inline loader"
          />
          <div
            className="ui divided relaxed list"
            role="list"
          />
          <button
            id="handle-prev"
            onClick={[Function]}
            type="button"
          >
            Prev
          </button>
          <button
            id="handle-next"
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
